### PR TITLE
Add a 'how do I...' section into docs

### PR DESCRIPTION
**Pull request by strickvl**
Adds a section with pointers to documentation for commonly-asked questions.

(Also (re)-adds the dependency clashes documentation + removes (unpublished) NLP project template link from docs.)

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/reference/how-do-i.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+`mlstacks` under the hood) which you can also use as a standalone solution if
+you are familiar with Terraform.
+
+- **deploy ZenML** on my internal company cluster?
+
+Read [the documentation on self-hosted ZenML
+deployments](https://docs.zenml.io/deploying-zenml/zenml-self-hosted) in which
+several options are presented.
+
+- **dynamic pipelines and steps**?
```
Comment (fa9r) [inline docs/book/reference/how-do-i.md:61]: I don't think users will know what we mean by "dynamic pipelines and steps"

File: docs/book/reference/how-do-i.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+on implementing this is the place to learn more.
+
+- **reset** things when something goes wrong?
+
+To reset your ZenML client, you can run `zenml clean` which will wipe your local
+metadata database and reset your client. Note that this is a destructive action,
+so feel free to [reach out to us on Slack](https://zenml.io/slack-invite/)
+before doing this if you are unsure.
+
+- **templates**: using starter code with ZenML?
```
Comment (fa9r) [inline docs/book/reference/how-do-i.md:83]: Whoever reads this docs page probably doesn't need starter templates anymore

Comment (fa9r) [reply docs/book/reference/how-do-i.md:83]: Unless they want to make templates for others I suppose

File: docs/book/reference/how-do-i.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+- **upgrade** my ZenML client and/or server?
+
+Upgrading your ZenML client package is as simple as running `pip install
+--upgrade zenml` in your terminal. For upgrading your ZenML server, please refer
+to [the dedicated documentation
+section](../deploying-zenml/zenml-self-hosted/manage-the-deployed-services/upgrade-the-version-of-the-zenml-server)
+which covers most of the ways you might do this as well as common
+troubleshooting steps.
+
+- use a <YOUR_FAVORITE_COMPONENT_GOES_HERE> stack component?
```
Comment (fa9r) [inline docs/book/reference/how-do-i.md:99]: You have a favourite component? 😄 

> Man I really love artifact stores, they are so much better than other stack components

File: docs/book/user-guide/advanced-guide/environment-management/handling-dependencies.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+requirements. We try to keep these dependency requirements ranges as wide as
+possible for the integrations developed by ZenML, but it is not always possible
+to make this work completely smoothly. Here are some of the known issues:
+
+- `label_studio`: the packages required for this integration are not compatible
+  with quite a few of our other integrations. At a minimum, you will get
+  warnings when installing `label_studio` via `zenml integration install ...`.
+  We have found these warnings to be mostly harmless. You might find you need to
+  separate your installations of Label Studio and any other integrations you
+  need for your environment / pipeline, however. Note [the warning on our own documentation page](../../../stacks-and-components/component-guide/annotators/label-studio.md#how-to-deploy-it)
```
Comment (fa9r) [inline docs/book/user-guide/advanced-guide/environment-management/handling-dependencies.md:59]: Line break

**Review by fa9r — APPROVED** [review]
Two very nice pages, let's get them live 🙌

File: docs/book/user-guide/starter-guide/using-project-templates.md [hunk 5]
```diff
@@ -39,7 +39,7 @@ What would you need to get a quick understanding of the ZenML framework and star
                 This project template is a good starting point for anyone starting with ZenML. It consists of two pipelines with the following high-level steps: load, split, and preprocess data; run HP tuning; train and evaluate model performance; promote model to production; detect data drift; run batch inference.
             </td>
         </tr>
-        <tr>
+        <!-- <tr>
```
Comment (strickvl) [inline docs/book/user-guide/starter-guide/using-project-templates.md:42]:
```suggestion
        <tr>
```

File: docs/book/user-guide/starter-guide/using-project-templates.md [hunk 6]
```diff
@@ -49,7 +49,7 @@ What would you need to get a quick understanding of the ZenML framework and star
             <td>
                 This project template is a simple NLP training pipeline that walks through tokenization, training, HP tuning, evaluation and deployment for a BERT or GPT-2 based model and testing locally it with gradio
             </td>
-        </tr>
+        </tr> -->
```
Comment (strickvl) [inline docs/book/user-guide/starter-guide/using-project-templates.md:52]:
```suggestion
        </tr>
```

**Pull request merged**